### PR TITLE
Resolve issues with data types in `cqcp_m5`, additional input check, `:=`

### LINKED ISSUE #29: `cqcp_m5()`: fails with spatial buddies in range
Unfortunately, running `cqcp_qcCWS()` on a Netatmo data sample fails currently with:

```
[CrowdQC+] QC level m5 could not meaningfully be performed with current configuration 
(only isolated stations with too few buddies). All flags m5 = FALSE. 
Consider increasing the radius, decreasing the number of buddies, or setting 'keep_isolated = TRUE'.
```

This does not seem to be correct since there are a lot of neighbouring stations nearby but the call even fails with e.g. `m5_radius = 10000` and `m5_n_buddies = 1`. 

netatmo_essen

Data can temporarily be accessed at [falk-env/share](https://github.com/falk-env/share/tree/main/data) and consists of:

- stations.rda - sf object containing stations and some metadata, but most importantly locations.
- dt_pad.rda - data.table according to requirements after `cqcp_padding()`, passes `cqcp_check_input()` without issues.

Is this a layer 8 issue or something not working properly with distances here? Thanks!

### PR DESCRIPTION
**Pull request by dafenner**
Resolves issues with data types of column "p_id" in `cqcp_m5`.
Includes a new check in `cqcp_check_input` regarding data types of the columns.
Resolves #29, #30, #31, #32, #33.

### CODE REVIEW

File: R/cqcp_filter.R [hunk 1]
```diff
@@ -369,14 +369,20 @@ cqcp_m5 <- function(data, radius = 3000, n_buddies = 5, alpha = 0.1,
   dist <- dist[(p_x != p_y) & (!is.na(distance))] # reduce table size
   combi <- dist[distance <= radius] # station within radius
   
+  # check for p_id column type (integer?) and convert indices to integer, if necessary
+  if (is.integer(data$p_id)) {
```
Comment (jkittner) [inline R/cqcp_filter.R:373]: isn't this supposed to be `if (!is.integer(data$p_id))` or what is `p_x` or `p_y` exactly?

File: R/cqcp_helper.R [hunk 2]
```diff
@@ -56,11 +56,12 @@ cqcp_colourise <- function(str, colour) {
 cqcp_check_input <- function(data, print = TRUE, file = NULL, as_list = FALSE){
   
   ok <- TRUE
-  ch_1a <- ch_1b <- ch_2 <- ch_3 <- ch_4 <- ch_5 <- TRUE
+  ch_1a <- ch_1b <- ch_1c <- ch_2 <- ch_3 <- ch_4 <- ch_5 <- TRUE
+  #mess_1a <- mess_1b <- mess_1c <- mess_2 <- mess_3 <- mess_4 <- mess_5 <- cqcp_colourise(paste0("     ! Check not carried out.\n"), "red")
```
Comment (jkittner) [inline R/cqcp_helper.R:60]: Is this supposed to be commented out? Might as well removed it then...

File: R/cqcp_helper.R [hunk 3]
```diff
@@ -163,7 +191,7 @@ cqcp_check_input <- function(data, print = TRUE, file = NULL, as_list = FALSE){
     ch_5 <- FALSE
     ok <- FALSE
   }
-  
+
```
Comment (jkittner) [inline R/cqcp_helper.R:194]:
```suggestion

```

**Review by jkittner — COMMENTED** [review]
looking great, just one small question (above)

File: R/cqcp_filter.R (hunk 1 above)
Comment (dafenner) [reply R/cqcp_filter.R:373]: No, actually correct. If `data$p_id` is of type integer, the two indices `p_x` and `p_y` need to be integer as well. Yet they are are of type character originally.

File: R/cqcp_helper.R (hunk 2 above)
Comment (dafenner) [reply R/cqcp_helper.R]: Thanks! Was included during development. Removed now.

**Pull request merged**